### PR TITLE
PL: Remove pegasus-based survey result viewing

### PR DESCRIPTION
**Pull request by breville**
This removes viewing survey results for pre-5/8/20 surveys which were being stored in Pegasus.

It removes:

- the "Facilitator Survey Results" and "Facilitator Survey Results" buttons from https://studio.code.org/pd/workshop_dashboard
- the https://studio.code.org/pd/workshop_dashboard/survey_results/[optional_workshop_id] and https://studio.code.org/pd/workshop_dashboard/organizer_survey_results/[optional_workshop_id] pages 
- the https://studio.code.org/api/v1/pd/workshops/[workshop_id]/workshop_survey_report
the https://studio.code.org/api/v1/pd/workshop_organizer_survey_report_for_course/[course] APIs
- the `Api::V1::Pd::WorkshopScoreSummarizer` module
- the `SurveyResults` and `OrganizerSurveyResults` React components
- the "View Survey Results" button in the main workshop dashboard table when it would have pointed at these legacy results pages.

#### The first button here is now gone:

Screenshot 2019-06-04 22 03 36

#### And this whole page is now gone:

Screenshot 2019-06-04 22 04 00

### CODE REVIEW

**Review by islemaster — CHANGES_REQUESTED** [review]
boom_baby

Awesome!  Two concerns:

1. Drone failures look relevant.
2. How much info do we have on whether anybody still uses these views?

**Review by agealy — DISMISSED** [review]
I'd like to pause on this until we've confirmed with ed team that they're fine with the change and given some thought to whether this warrants any other communication to the user about old results. I can own following up

**Comment by hacodeorg** [comment]
Adding 1 more place to remove references to `survey_results` and `organizer_survey_results` https://github.com/code-dot-org/code-dot-org/blob/c65dcd2660f317d44026e48e6c66e1a3462e6db1/apps/src/code-studio/pd/workshop_dashboard/components/workshop_management.jsx#L44-L46

**Comment by hacodeorg** [comment]
Fun fact, I was thinking about what if we need the knowledge in these files in the future, and then I saw [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) (You aren't gonna need it) principle from extreme programming. It's loosely relevant to removing dead code, but still makes sense.

**Review by islemaster — APPROVED** [review]
Looks great! Whenever we get approval from stakeholders...

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_management.jsx [hunk 1]
```diff
@@ -59,11 +59,14 @@ export class WorkshopManagement extends React.Component {
         this.props.subject
       );
 
+    // 2018-08-01 is when we started using JotForm.  Don't change this date here.
     let new_facilitator_weekend =
       workshop_date >= new Date('2018-08-01') &&
       ['CS Discoveries', 'CS Principles'].includes(this.props.course) &&
       this.props.subject !== 'Code.org Facilitator Weekend';
 
+    // 2019-05-20 is when we started using a JotForm survey for Deep Dive workshops.
```
Comment (hacodeorg) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_management.jsx:79]: Thank you :D

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_managementTest.js [hunk 2]
```diff
@@ -117,20 +117,20 @@ describe('WorkshopManagement', () => {
     it('uses organizer results for organizers', () => {
```
Comment (hacodeorg) [inline apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_managementTest.js:117]: Should we delete these tests? I'm not sure what their purposes are now.

**Review by hacodeorg — APPROVED** [review]

File: apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_managementTest.js (hunk 2 above)
Comment (breville) [reply apps/test/unit/code-studio/pd/workshop_dashboard/components/workshop_managementTest.js:117]: Good question.  I'm neutral.  For what it's worth, the `getSurveyUrlForProps` function still returns valid values in some tests, and so these tests do exercise the removals in this PR.

**Comment by breville** [comment]
Revisiting this exactly a year later with a new merge from staging and lots of help resolving conflicts from @molly-moen.  We believe we can now go ahead with this change since the CSF Intro survey [now uses Foorm](https://github.com/code-dot-org/code-dot-org/pull/34685), and we [now have legacy facilitator survey summaries](https://github.com/code-dot-org/code-dot-org/pull/34654).

**Review by molly-moen — APPROVED** [review]
🎉

**Review by islemaster — APPROVED** [review]
:tada: :tada: :tada:

**Review by islemaster — APPROVED** [review]

**Pull request merged**